sem_covid/entrypoints/etl_dags/finreg.py

```diff
@@ -0,0 +1,282 @@
+        ev:4279
+    }
+    VALUES ?resource_type {
+        res_type:REG
+        fd_030:REGL
+        res_type:DIR
+        res_type:REG_IMPL
+        res_type:REG_DEL
+        fd_030:DIRECTIVE
+        fd_030:REGIMP
+        fd_030:REGDEL
+        res_type:DIR_DEL
+        res_type:DIR_IMPL
+        res_type:REG_FINANC
+        fd_030:DIRDEL
+        fd_030:DIRIMP
+        fd_030:DIR
+        fd_030:REG
+    }
+
+    ?work cdm:work_is_about_concept_eurovoc ?eurovoc_concept ;
+        cdm:work_has_resource-type|cdm:resource_legal_has_type_act_concept_type_act ?resource_type .
+    OPTIONAL {
+        ?exp cdm:expression_belongs_to_work ?work ;
+            cdm:expression_uses_language ?expr_lang ;
+            cdm:expression_title ?_title .
+    }
+}
+GROUP BY ?work'''
+
+
+def make_request(query):
+    wrapper = SPARQLWrapper(config.EU_CELLAR_SPARQL_URL)
+    wrapper.setQuery(query)
+    wrapper.setReturnFormat(JSON)
+
+    return wrapper.query().convert()
+
+
+def get_single_item(query, json_file_name):
+    minio = StoreRegistry.minio_object_store(config.EU_FINREG_CELLAR_BUCKET_NAME)
+    response = make_request(query)['results']['bindings']
+    uploaded_bytes = minio.put_object(json_file_name, json.dumps(response).encode('utf-8'))
+    logger.info(f'Save query result to {json_file_name}')
+    logger.info('Uploaded ' +
+                str(uploaded_bytes) +
+                ' bytes to bucket [' +
+                config.EU_FINREG_CELLAR_BUCKET_NAME + '] at ' +
+                config.MINIO_URL)
+    return response
+
+
+def download_and_split_callable():
+    minio = StoreRegistry.minio_object_store(config.EU_FINREG_CELLAR_BUCKET_NAME)
+    minio.empty_bucket(object_name_prefix=None)
+    minio.empty_bucket(object_name_prefix=RESOURCE_FILE_PREFIX)
+    minio.empty_bucket(object_name_prefix=TIKA_FILE_PREFIX)
+    minio.empty_bucket(object_name_prefix=FIELD_DATA_PREFIX)
+
+    get_single_item(QUERY, config.EU_FINREG_CELLAR_JSON)
+
+
+def execute_worker_dags_callable(**context):
+    minio = StoreRegistry.minio_object_store(config.EU_FINREG_CELLAR_BUCKET_NAME)
+    works = json.loads(minio.get_object(config.EU_FINREG_CELLAR_JSON).decode('utf-8'))
+    count = 0
+
+    for work in works:
+        TriggerDagRunOperator(
+            task_id='trigger_slave_dag_finreg_' + str(count),
+            trigger_dag_id=SLAVE_DAG_NAME,
+            conf={"work": work['work']['value']}
+        ).execute(context)
+        count += 1
+
+    logger.info("Created " + str(count) + " DAG runs")
+
+
+default_args = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2021, 2, 22),
+    "email": ["mclaurentiu79@gmail.com"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "retry_delay": timedelta(minutes=3600)
+}
+with DAG(DAG_NAME, default_args=default_args, schedule_interval="@once", max_active_runs=1, concurrency=4) as dag:
+    download_task = PythonOperator(task_id='download_and_split',
```

Comment:
can let's add retry to this step, as it may fail when querying Cellar.

Reply:
yes. keep in mind that any retry occupy a slot in the pool, and if retrying tasks are occupying the pool, no new tasks will run until slots in the pool become available.